### ISSUE TITLE
Invalid client id or redirect uri

### ISSUE THREAD

**Issue by Vinz87** [home-assistant/home-assistant-iOS#468]
I get this error when trying to connect with the iOS app.

I only left "default_config:" in my configuration.yaml.

I put the local address in Home Assistant. If I open that address in Safari, I can get the login form without problems, while in the iOS app results that error.

Home Assistant installed on a Raspberry Pi, with a Python2.7 virtual environment.

Home Assistant v0.100.3, Python3.7 virtual environment on a Raspberry Pi

**Comment by TomBrien** [comment]
Possible fixes are:
- Delete all access tokens and try again (may take a few goes)
- Make sure you have something like the following in your `configuration.yaml` (note no protocol or port):
  ```YAML
  http:
    base_url: domain.org
  ```

**Comment by Vinz87** [comment]
I forgot to add that it worked until some point, so it shouldn't be something outside Home Assistant the cause, since I didn't modified anything since it was working. 

I had the base_url set to my ddns address, I removed to see if it was the problem and kept trying only on my local network.

I'll try the access token cleaning, is it from three Settings panel in Home assistant right?

**Comment by TomBrien** [comment]
If it was working and nothing changed I would guess towards the access tokens. You can find them under your user profile in Home Assistant.

**Comment by Vinz87** [comment]
Deleted all access tokens (except the first of the list which can not be deleted, which should be the one from which I'm currently logged on my macbook), deleted and reinstalled Home Assistant on the iPhone.
It automatically sees the Raspberry Pi's IP, but trying to connect I get the same error.

**Comment by Vinz87** [comment]
This is the URL that the iOS tries to open that gives problems:
http://HOME_ASSISTANT_IP:8123/auth/authorize?response_type=code&client_id=https://home-assistant.io/iOS&redirect_uri=homeassistant://auth-callback

Again, accessing from the iPhone through the normal URL http://HOME_ASSISTANT_IP:8123 gives no problems.

**Comment by Vinz87** [comment]
After opening the normal URL http://HOME_ASSISTANT_IP:8123 I get this redirected URL: 
HOME_ASSISTANT_IP:8123/auth/authorize?response_type=code&redirect_uri=http%3A%2F%2FHOME_ASSISTANT_IP%3A8123%2F%3Fauth_callback%3D1&client_id=http%3A%2F%2FHOME_ASSISTANT_IP%3A8123%2F&state=eyJoYXNzVXJsIjoiaHR0cDovLzE5Mi4xNjguMTc4LjI2OjgxMjMiLCJjbGllbnRJZCI6Imh0dHA6Ly8xOTIuMTY4LjE3OC4yNjo4MTIzLyJ9

which correctly shows the login form.
So why is this differenent from what I get from the iOS app?

**Comment by Vinz87** [comment]
I tried to uninstall Home Assistant, and on a fresh install of v0.101.3 I get the same error.
Can't access from the app on both local and DDNS address. Both of them work correctly if accessed from the browser.

**Comment by Vinz87** [comment]
The only thing that worked was to downgrade iOS app to v1.1.1.
Current v1.5.1 and beta v2.0 give this problem.

**Comment by Vinz87** [comment]
After trying to completely disable IPv6 from my network as suggested by the app developer on the Discord channel, without solving the issue, I finally flashed the last version of Raspbian Buster on the Raspberry Pi.
On top of that brand new OS I followed the manual installation instructions (which means I also upgraded to HA v0.102.0), and I can report the issue disappeared on the v2.0beta and v1.5.1 iOS app.

I still don't know the reason of the problem but I consider the issue closed.
